### PR TITLE
add `service-name` to java.json

### PR DESCRIPTION
**Pull request by XiaofeiCao**
fix https://github.com/Azure/autorest.java/issues/1656

add 
```json 
{
  "service-name": "My Service Name"
}
``` 
to `java.json`

or `SERVICE_NAME` to environment variable will enable this feature.

tested locally

### CODE REVIEW

File: cadl-extension/src/main/java/com/azure/autorest/CadlPlugin.java [hunk 1]
```diff
@@ -142,6 +148,7 @@ public CadlPlugin(Options options) {
         if (!CoreUtils.isNullOrEmpty(options.outputFolder)) {
             SETTINGS_MAP.put("output-folder", options.outputFolder);
         }
+        SETTINGS_MAP.put("service-name", options.serviceName);
```
Comment (weidongxu-microsoft) [inline cadl-extension/src/main/java/com/azure/autorest/CadlPlugin.java:151]: null/empty check?

**Review by weidongxu-microsoft — APPROVED** [review]

File: cadl-extension/src/main/java/com/azure/autorest/CadlPlugin.java (hunk 1 above)
Comment (XiaofeiCao) [reply cadl-extension/src/main/java/com/azure/autorest/CadlPlugin.java]: Added.

**Comment by weidongxu-microsoft** [comment]
refresh main

**Comment by XiaofeiCao** [comment]
Seems like I had to manually add the diff. CI's experiencing concurrent issue here:
1. [PartialUpdate PR](https://github.com/Azure/autorest.java/pull/1653) passed CI.
2. [LRO PR](https://github.com/Azure/autorest.java/pull/1652) is raised. It got [something](https://github.com/Azure/autorest.java/pull/1652/files#diff-faa81291b1b3a505d0ad3ebebf64e229ba2cffdb5639138b735ada91b3abea7cR138-R140) that would be changed by [PartialUpdate PR](https://github.com/Azure/autorest.java/pull/1653). It passed CI too.
3. [LRO PR](https://github.com/Azure/autorest.java/pull/1652) got merged.
4. [PartialUpdate PR](https://github.com/Azure/autorest.java/pull/1653) got merged after.

This would leave the project in an inconsistent state.

**Review by haolingdong-msft — APPROVED** [review]

**Pull request merged**